kickstarter/android-oss
# NT-1648: Update copy and fix interstitial screen copy

# 📲 What

- Update "Open Inbox" copy on email verification interstitial screen to be sentence case instead of title case.
- Fix but caused by deprecated `getUser` method from User object class. 

# 🤔 Why

Upon added a required `user` parameter to the interstitial fragment for the login and signup activity, opening the interstitial from the internal tools screen was crashing because the user object was being called by a deprecated method. The user object was null and it caused a `nullpointerexpection`.

# 🛠 How

In response to clicking the Interstitial screen button in the internal tools page, the activity subscribes to an observable provided by the `user` model that emits the current user. If this stream emits, the fragment is started. This means, if you are not logged in, you will not be able to see the interstitial screen from the internal tools page. 

# 👀 See

This is an example of the copy change I made: 
Before and After
![Screenshot_1604659516](https://user-images.githubusercontent.com/19390326/99111316-e076ed80-25b9-11eb-9a72-904ece253db8.png) ![Screenshot_1605294349](https://user-images.githubusercontent.com/19390326/99111292-d7861c00-25b9-11eb-9df5-ceab5d2b776a.png)

# 📋 QA

Crash QA:
- Tap the overflow menu at the top left of the screen
- Select "Internal Tools"
- Scroll to the bottom and select "Email Verification Interstitial" 
        - Not logged in: There should be no reaction.
        - Logged in: The interstitial screen should appear

Copy QA:
- Make sure you are logged in and repeat the steps above
- Observe that the interstitial cta button says "Open inbox", in sentence case. 

# Story 📖

[Build interstitial UI](https://kickstarter.atlassian.net/browse/NT-1648)
